CONTRIBUTING.md

```diff
@@ -0,0 +1,119 @@
+See [this documentation](docs/custom_fields.md#via-code) on adding a new recognizer via code. The [PatternRecognizer](presidio-analyzer/analyzer/pattern_recognizer.py) class should be extended.
+See some examples here:
+  - [Credit card recognizer](presidio-analyzer/analyzer/predefined_recognizers/credit_card_recognizer.py)
+  - [Email recognizer](presidio-analyzer/analyzer/predefined_recognizers/email_recognizer.py)
+
+3. Machine Learning (ML) based or rule-based
+
+Many PII entites are undetectable using naive approaches like black-lists or regular expressions. In these cases, we would wish to utilize a Machine Learning model capabable of identifying entities in text, or a rule-based recognizer.
+There are four options for adding ML and rule based recognizers:
+
+   - Utilize spaCy:
+
+   Presidio currently uses [spaCy](https://spacy.io/) as a framework for text analysis and Named Entity Recognition (NER). As we prefer to use the existing tools, it would be best to contribute ML models that are based on spaCy. spaCy provides descent results compared to state-of-the-art NER models, but with much better computational performance. spaCy could be trained from scratch, used in combination with pre-trained embeddings, or retrained to detect new entities. When building such model, you would need to extend the [EntityRecognizer](presidio-analyzer/analyzer/entity_recognizer.py) class.
+   
+   - Utilize Scikit-learn or similar
+   
+   Scikit-learn models tend to be fast, but usually have lower accuracy than deep learning methods. However, for well defined problems with well defined features, they can provide very good results.
+   Since deep learning models tend to be complex and slow, we encourage you to first test a traditional approach (like Conditional Random Fields) before going directly into state-of-the-art *Sesame-Street* character based models... 
+   When building such model, you would need to extend the [EntityRecognizer](presidio-analyzer/analyzer/entity_recognizer.py) class.
+
+   - Use custom logic
+
+   In some cases, rule-based logic provides the best way of detecting entities. The Presidio EntityRecognizer API allows you to use spaCy extracted features like lemmas, part of speech, dependencies and more to create your logic. When building such model, you would need to extend the [EntityRecognizer](presidio-analyzer/analyzer/entity_recognizer.py) class.
+
+   - Deep learning based methods
+
+   Deep learning methods offer excellent detection rates for NER. They are however more complex to train, deploy and tend to be slower than traditional approaches. When contributing a DL based method, the best option would be to create a sidecar container which is isolated from the presidio-analyzer container. On the `presidio-analyzer` side, one would extend the [RemoteRecognizer](presidio-analyzer/analyzer/remote_recognizer.py) class and implement the network interface between `presidio-analyzer` and the endpoint of the model's container.
+
```

Comment:
Great explanation. Also add that any contribution to the supported types should be added here:
https://github.com/microsoft/presidio/blob/master/docs/field_types.md

Reply:
Added